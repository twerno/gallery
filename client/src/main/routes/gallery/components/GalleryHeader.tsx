import Button from 'main/components/styled/Button';
import Input from 'main/components/styled/Input';
import { Field, Formik } from 'formik';
import * as React from 'react';
import Headroom from 'react-headroom';
import { IGalleryUrlQuery } from 'main/routes/Path';

import HeaderSearchForm from './styled/HeaderSearchForm';

export interface IGalleryHeaderProps {
    query: IGalleryUrlQuery;
    onSearchSubmitted: (query: IGalleryUrlQuery) => void;
}

export const GalleryHeader = (props: IGalleryHeaderProps) => {

    return (
        <Headroom style={{ backgroundColor: 'white', zIndex: 10 }}>

            <Formik
                initialValues={{ q: props.query.q }}
                onSubmit={
                    async (values, { setSubmitting }) => {
                        props.onSearchSubmitted({ q: values.q });
                    }
                }
            >
                {
                    ({ isSubmitting }) => (
                        <HeaderSearchForm>
                            <Field name="q" as={Input} aria-label="Search" />
                            <Button type="submit" disabled={isSubmitting} disableLeftBorder={true}>
                                Search
                            </Button>
                        </HeaderSearchForm>
                    )
                }
            </Formik>

        </Headroom >
    );
}