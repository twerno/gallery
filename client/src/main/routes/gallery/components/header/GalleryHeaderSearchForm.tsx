import { useFormik } from 'formik';
import StyledButton from 'main/components/styled/StyledButton';
import StyledInput from 'main/components/styled/StyledInput';
import * as React from 'react';
import { IGalleryUrlQuery } from '../../model/galleryQuery';
import GalleryHeaderStyles from './GalleryHeaderStyles';

export interface IGalleryHeaderSearchFormProps {
    initialValues: IGalleryUrlQuery;
    onSubmit: (query: IGalleryUrlQuery) => void;
}

export const GalleryHeaderSearchForm = ({ initialValues, onSubmit }: IGalleryHeaderSearchFormProps) => {

    const { handleChange, handleSubmit, values, isSubmitting } = useFormik({
        initialValues,
        onSubmit
    });

    return (
        <GalleryHeaderStyles.SearchForm onSubmit={handleSubmit}>

            <StyledInput
                id="q"
                name="q"
                aria-label="Search"
                type="text"
                onChange={handleChange}
                value={values.q || ''}
            />

            <StyledButton
                mode='inputButton'
                type="submit"
                disabled={isSubmitting}
            >
                Search
            </StyledButton>

        </GalleryHeaderStyles.SearchForm>
    );
}