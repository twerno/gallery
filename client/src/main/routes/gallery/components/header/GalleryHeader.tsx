import * as React from 'react';
import Headroom from 'react-headroom';
import { GalleryHeaderSearchForm, IGalleryHeaderSearchFormProps } from './GalleryHeaderSearchForm';

export interface IGalleryHeaderProps extends IGalleryHeaderSearchFormProps {

}

export const GalleryHeader = (props: IGalleryHeaderProps) => {

    return (
        <Headroom style={{ backgroundColor: 'white', zIndex: 10 }}>
            <GalleryHeaderSearchForm {...props} />
        </Headroom>
    );
}