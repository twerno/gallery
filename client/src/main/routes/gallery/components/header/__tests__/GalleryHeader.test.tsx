import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import * as React from 'react';
import { componentBasicTests } from 'test/componentBasicTests';
import { GalleryHeader, IGalleryHeaderProps } from '../GalleryHeader';
import { GalleryHeaderSearchForm } from '../GalleryHeaderSearchForm';

describe('GalleryHeader', () => {

    const props: IGalleryHeaderProps = {
        initialValues: { q: 'cats' },
        onSubmit: (_) => { },
    };

    componentBasicTests(
        "GalleryHeader",
        <GalleryHeader {...props} />
    );

    test("passing props", async () => {
        const wrapper = shallow(<GalleryHeader {...props} />);
        const searchForm = wrapper.find(GalleryHeaderSearchForm);

        expect(searchForm.props().initialValues).toBe(props.initialValues);
        expect(searchForm.props().onSubmit).toBe(props.onSubmit);
    });

});

