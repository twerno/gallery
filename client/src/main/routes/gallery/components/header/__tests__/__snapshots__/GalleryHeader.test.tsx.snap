// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryHeader [GalleryHeader] Component basic tests matches the snapshot 1`] = `
<Headroom
  calcHeightOnResize={true}
  disable={false}
  disableInlineStyles={false}
  downTolerance={0}
  onPin={[Function]}
  onUnfix={[Function]}
  onUnpin={[Function]}
  parent={[Function]}
  pinStart={0}
  style={
    Object {
      "backgroundColor": "white",
      "zIndex": 10,
    }
  }
  upTolerance={5}
  wrapperStyle={Object {}}
>
  <Component
    initialValues={
      Object {
        "q": "cats",
      }
    }
    onSubmit={[Function]}
  />
</Headroom>
`;
